php/PHPFileInfo/StringBasedPHPFileInfo.php

```diff
@@ -0,0 +1,456 @@
+<?php
+
+namespace PHPCD\PHPFileInfo;
+
+use SplFileObject;
+
+/**
+ *
+ */
+class StringBasedPHPFileInfo implements PHPFileInfo
+{
+    const USE_PATTERN =
+        '/^use\s+((?<type>(constant|function)) )?(?<left>[\\\\\w]+\\\\)?({)?(?<right>[\\\\,\w\s]+)(})?\s*;$/';
+
+    const ALIAS_PATTERN = '/(?<suffix>[\\\\\w]+)(\s+as\s+(?<alias>\w+))?/';
+
+    const CLASS_PATTERN = '/^\s*\b((((final|abstract)\s+)?class)|interface|trait)\s+(?<class>\S+)/i';
+
+    const EXTENDS_PATTERN = '/\s*\bextends\s+(?<superclass>\S+)\b/i';
+
+    const IMPLEMENTS_PATTERN = '/\s*\bimplements\s+(?<interfaces>(\w+(,\s+)?)+)/i';
+
+    const NAMESPACE_PATTERN = '/(<\?php)?\s*namespace\s+(?<namespace>.*);$/';
+
+    /*
+     * @var SplFileObject
+     */
+    private $file;
+
+    private $namespace = '';
+
+    private $class;
+
+    private $superclass;
+
+    private $interfaces = [];
+
+    private $imports = [];
+
+    private $errors;
+
+    public function __construct(SplFileObject $file)
+    {
+        try {
+            $this->file = $file;
+
+            $this->validateSyntax();
+
+            $this->scanFile();
+
+            if (!empty($this->getSuperclass())) {
+                $this->classExists($this->getSuperclass());
```

Comment:
See `tests/php/Fixtures/StringBasedPHPFileInfo/ExampleWithNonExistingSuperclass.php`. It condition comes from real example.

If you make in `php` class
```
<?php

namespace PHPCD;

class A extends X
{
}
```
you probably will get something like:
```
WARNING: 2016-07-20 01:23:21 Class PHPCD\A did not passed validation and then cannot be added to class info repository. Reason:
[
    "Class PHPCD\\X does not exist."
]
WARNING: 2016-07-20 01:23:22 Class PHPCD\Fixtures\StringBasedPHPFileInfo\A did not passed validation and then cannot be added to class info repository. Reason:
[
    "Class A\\X1234 does not exist."
]
```

But thanks for attention, I found bug when interface extends another interface and when class extends something described as path.